torch/_dynamo/optimizations/training.py

```diff
@@ -86,107 +151,28 @@ def raise_or_warn(reason):
     return True
 
 
-class AotAutogradStrategy(object):
-    """Base class for backend strategies that use AOT Autograd"""
-
-    @classmethod
-    def compile_fn(cls, gm: torch.fx.GraphModule, example_inputs):
-        if count_calls(gm.graph) < 2:
-            return gm  # no point for tiny graphs
-        return cls(gm, example_inputs).verified_candidate()
-
-    def __init__(self, gm: torch.fx.GraphModule, example_inputs):
-        import functorch.compile
-
-        functorch.compile.config.use_functionalize = True
-        functorch.compile.config.use_fake_tensor = True
-
-        super(AotAutogradStrategy, self).__init__()
-        counters["aot_autograd"]["total"] += 1
-        self.use_fallback = False
-        self.original_example_inputs = example_inputs
-        self.gm = gm
-
-        if not functorch.compile.config.use_functionalize and config.normalize_ir:
-            try:
-                self.gm = normalize_ir(gm, self.example_inputs)
-            except Exception:
-                log.debug("TorchDynamo unable to remove mutation")
-                self.use_fallback = True
-                pass
-
-        if not is_aot_autograd_safe_to_run(gm, example_inputs):
-            self.use_fallback = True
-
-    @property
-    def example_inputs(self):
-        return clone_inputs(self.original_example_inputs)
-
-    def verified_candidate(self):
-        if self.use_fallback:
-            log.debug("Unable to use AOT Autograd because graph has mutation")
-            counters["aot_autograd"]["not_ok"] += 1
-            return self.gm
-        cg = self.candidate()
-        if cg is None:
-            counters["aot_autograd"]["not_ok"] += 1
-            raise RuntimeError("AOT Autograd failed to compile")
-        counters["aot_autograd"]["ok"] += 1
-        return cg
-
-    def candidate(self):
-        raise NotImplementedError()
-
-
-class AotNop(AotAutogradStrategy):
-    """Useful for debugging purpose"""
-
-    def candidate(self):
-        from functorch._src.compilers import debug_nop
-        from functorch.compile import nop
-
-        DEBUG = False
-        return BACKENDS["aot_autograd"](
-            self.gm, self.example_inputs, fw_compiler=debug_nop if DEBUG else nop
-        )  # type: ignore[call-arg]
-
-
-aot_eager = AotNop.compile_fn
-
-
-class AotTorchscript(AotAutogradStrategy):
-    """
-    AOT Autograd with torchscript backend. Default partitioner.
-    """
-
-    def candidate(self):
-        from functorch.compile import ts_compile
-
-        return BACKENDS["aot_autograd"](
-            self.gm, self.example_inputs, fw_compiler=ts_compile
-        )  # type: ignore[call-arg]
-
-
-aot_ts = AotTorchscript.compile_fn
+DEBUG = False
```

Comment:
Should this be in the config?

Reply:
Yes, probably. Need to negotiate a name for it.